src/Controller/EnMarche/CommitteeController.php

```diff
@@ -48,13 +50,80 @@ public function showAction(Request $request, Committee $committee): Response
+                'committee_timeline' => $committeeManager->getTimeline($committee,
+                    $this->getParameter('timeline_max_messages')),
+                'committee_timeline_max_messages' => $this->getParameter('timeline_max_messages'),
+                'form' => $form ? $form->createView() : null,
+            ]);
+    }
+
+    /**
+     * @Route("/timeline/{id}/edit", name="app_committee_timeline_edit")
+     * @ParamConverter("committee", options={"mapping":{"slug": "slug"}})
+     * @ParamConverter("committeeFeedItem", options={"mapping":{"id": "id"}})
+     * @Method("GET|POST")
+     * @Security("is_granted('SHOW_COMMITTEE', committee)")
+     */
+    public function timelineEditAction(Request $request, Committee $committee, CommitteeFeedItem $committeeFeedItem): Response
+    {
+        if (!($user = $this->getUser()) instanceof Adherent
+            || ($committeeFeedItem->getAuthor() !== $user
+                && !$user->isSupervisorOf($committee))) {
+            throw $this->createAccessDeniedException('Vous n\'avez pas les droits pour editer cette resource.');
+        }
+        $form = null;
+        if ($this->isGranted(CommitteePermissions::HOST, $committee)) {
+            $form = $this
+                ->createForm(CommitteeFeedMessageType::class, $committeeFeedItem, ['data_class' => CommitteeFeedItem::class])
+                ->handleRequest($request)
+            ;
+
+            if ($form->isSubmitted() && $form->isValid()) {
+                $this->getDoctrine()->getManager()->flush();
+                $this->addFlash('info', $this->get('translator')->trans('committee.message_edited'));
+
+                return $this->redirect($this->generateUrl('app_committee_show', ['slug' => $committee->getSlug()]));
+            }
+        }
+
+        $committeeManager = $this->get('app.committee.manager');
+
+        return $this->render('committee/show.html.twig',
+            [
+                'committee' => $committee,
+                'committee_hosts' => $committeeManager->getCommitteeHosts($committee),
+                'committee_timeline' => $committeeManager->getTimeline($committee,
+                    $this->getParameter('timeline_max_messages')),
+                'committee_timeline_max_messages' => $this->getParameter('timeline_max_messages'),
+                'form' => $form ? $form->createView() : null,
+            ]);
+    }
+
+    /**
+     * @Route("/timeline/{id}/delete", name="app_committee_timeline_delete")
+     * @ParamConverter("committee", options={"mapping":{"slug": "slug"}})
+     * @ParamConverter("committeeFeedItem", options={"mapping":{"id": "id"}})
+     * @Method("GET")
+     * @Security("is_granted('SHOW_COMMITTEE', committee)")
+     */
+    public function timelineDeleteAction(Committee $committee, CommitteeFeedItem $committeeFeedItem): Response
+    {
+        if (!($user = $this->getUser()) instanceof Adherent
```

Comment:
Same here, you should use a Voter and for example : `@Security("is_granted('DELETE_TIMELINE, committee)")`

Reply:
What about a single Voter for `TIMELINE_MANAGE` then?

Reply:
What about an `AdministrateTimelineVoter` and an `ADMINISTRATE_TIMELINE` role ?